tests/infra_test.go

```diff
@@ -1052,8 +1058,109 @@ var _ = Describe("[Serial][sig-compute]Infrastructure", func() {
 			}
 		})
+		AfterEach(func() {
+			nodesWithKVM = tests.GetNodesWithKVM()
+
+			for _, node := range nodesWithKVM {
+				delete(node.Labels, nonExistingCPUModelLabel)
+
+				p := []patch{
+					{
+						Op:    "replace",
+						Path:  "/metadata/labels",
+						Value: node.Labels,
+					},
+				}
+
+				delete(node.Annotations, v1.LabellerSkipNodeAnnotation)
+
+				p = append(p, patch{
+					Op:    "replace",
+					Path:  "/metadata/annotations",
+					Value: node.Annotations,
+				})
+
+				payloadBytes, err := json.Marshal(p)
+				Expect(err).ToNot(HaveOccurred())
+
+				_, err = virtClient.CoreV1().Nodes().Patch(context.Background(), node.Name, types.JSONPatchType, payloadBytes, metav1.PatchOptions{})
+				Expect(err).ToNot(HaveOccurred())
+			}
+		})
 
 		Context("basic labelling", func() {
+			It("skip node reconciliation when node has skip annotation", func() {
+
+				for i, node := range nodesWithKVM {
+					node.Labels[nonExistingCPUModelLabel] = "true"
+					p := []patch{
+						{
+							Op:    "add",
+							Path:  "/metadata/labels",
+							Value: node.Labels,
+						},
+					}
+					if i == 0 {
+						node.Annotations[v1.LabellerSkipNodeAnnotation] = "true"
+
+						p = append(p, patch{
+							Op:    "add",
+							Path:  "/metadata/annotations",
+							Value: node.Annotations,
+						})
+					}
+					payloadBytes, err := json.Marshal(p)
+					Expect(err).ToNot(HaveOccurred())
+
+					_, err = virtClient.CoreV1().Nodes().Patch(context.Background(), node.Name, types.JSONPatchType, payloadBytes, metav1.PatchOptions{})
+					Expect(err).ToNot(HaveOccurred())
+				}
+				kvConfig := v1.KubeVirtConfiguration{ObsoleteCPUModels: map[string]bool{}}
+				// trigger reconciliation
+				tests.UpdateKubeVirtConfigValueAndWait(kvConfig)
+
+				Eventually(func() bool {
+					nodesWithKVM = tests.GetNodesWithKVM()
+
+					for _, node := range nodesWithKVM {
+						_, skipAnnotationFound := node.Annotations[v1.LabellerSkipNodeAnnotation]
+						_, customLabelFound := node.Labels[nonExistingCPUModelLabel]
+						if customLabelFound && !skipAnnotationFound {
+							return false
+						}
+					}
+					return true
+				}, 15*time.Second, 1*time.Second).Should(Equal(true))
+
+				nodesWithKVM = tests.GetNodesWithKVM()
+
+				for _, node := range nodesWithKVM {
+					delete(node.Labels, nonExistingCPUModelLabel)
+
+					p := []patch{
+						{
+							Op:    "replace",
+							Path:  "/metadata/labels",
+							Value: node.Labels,
+						},
+					}
+
+					delete(node.Annotations, v1.LabellerSkipNodeAnnotation)
+
+					p = append(p, patch{
+						Op:    "replace",
+						Path:  "/metadata/annotations",
+						Value: node.Annotations,
+					})
+
+					payloadBytes, err := json.Marshal(p)
+					Expect(err).ToNot(HaveOccurred())
+
+					_, err = virtClient.CoreV1().Nodes().Patch(context.Background(), node.Name, types.JSONPatchType, payloadBytes, metav1.PatchOptions{})
+					Expect(err).ToNot(HaveOccurred())
```

Comment:
This is cleanup right?  If so, the AfterEach function should be ablel to handle this.

Reply:
Yes, I forgot to remove it.